src/details/ArborX_DetailsVector.hpp

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_DETAILS_VECTOR_HPP
+#define ARBORX_DETAILS_VECTOR_HPP
+
+#include <ArborX_HyperPoint.hpp>
+
+namespace ArborX::Details
+{
+
+// Same as Point, but a different type (can't inherit from Point as it would
+// run into constexpr and inability to aggregate initialize)
+template <int DIM, typename Coordinate = float>
+struct Vector
+{
+  static_assert(DIM > 0);
+  // Initialization is needed to be able to use Point in constexpr
+  // TODO: do we want to actually want to zero initialize it? Seems like
+  // unnecessary work.
+  Coordinate _coords[DIM] = {};
+
+  KOKKOS_FUNCTION
+  constexpr auto &operator[](unsigned int i) { return _coords[i]; }
+
+  KOKKOS_FUNCTION
+  constexpr auto const &operator[](unsigned int i) const { return _coords[i]; }
+
+  friend KOKKOS_FUNCTION constexpr bool operator==(Vector const &v,
+                                                   Vector const &w)
+  {
+    bool match = true;
+    for (int d = 0; d < DIM; ++d)
+      match &= (v[d] == w[d]);
+    return match;
+  }
+};
+
+template <typename... T>
+Vector(T...) -> Vector<
+    sizeof...(T),
+    std::conditional_t<(... || std::is_same_v<std::decay_t<T>, double>), double,
+                       float>>;
```

Comment:
This looks like a bad idea.
Were you just trying to achieve `Vector{1,2,3}` interpreted as `Vector<3, float>` or did you really think through what this entails?

Reply:
I know what this logic does as I took it straight from hyper point. Yes, I indeed try to make it a `float` by default unless it's a `double`.